### PR TITLE
Feat/activity pending fallback

### PR DESCRIPTION
**Pull request by bigboydiamonds**
- Add hooks `useFallbackBridgeOriginQuery` `useFallbackBridgeDestinationQuery` Explorer fallback queries when Pending Transaction duration has exceeded calculated estimated transaction duration 
- Update `/transactions/updater.tsx` to handle pending/historical transactions with increased data context with fallback query responses
- Ensure de-duplication amongst `transactions` state that stores fall back transactions (compared to transactions grabbed from default Explorer endpoint query)
- Enable [Redux Persist](https://github.com/synapsecns/sanguine/blob/feat/activity-pending-fallback/packages/synapse-interface/store/reducer.ts) to allow for persisted state stored in local browser cache so state remains unchanged between refreshes
- Reset Bridge Inputs when address updated
- Implement feature to cache bust based on `persistConfig.version` to allow for data schema changes without breaking app in the future

## Summary by CodeRabbit

New Features:
- Introduced new GraphQL queries `GetDestinationBridgeTxFallback` and `GetOriginBridgeTxFallback` to retrieve bridge transaction information.
- Implemented new hooks `useFallbackBridgeOriginQuery` and `useFallbackBridgeDestinationQuery` to fetch fallback bridge origin and destination queries.
- Added Redux store persistence using `redux-persist` to ensure state is preserved across page reloads.

Improvements:
- Enhanced transaction handling by incorporating fallback queries for pending and historical transactions.
- Updated `BridgeExchangeRateInfo` component to ensure `safeExchangeRate` is always a `BigInt` value.
- Modified `TransactionPayloadDetail` component to handle `tokenAmount` as both string and number types.

Bug Fixes:
- Fixed a typo in the action type of `removeFallbackQueryHistoricalTransaction`.

Refactor:
- Optimized transaction state management by introducing new arrays `fallbackQueryPendingTransactions` and `fallbackQueryHistoricalTransactions`.
- Updated Redux store configuration to include new reducers and slices.

Tests:
- Added new functions `addFallbackQueryHistoricalTransaction`, `removeFallbackQueryHistoricalTransaction`, and `removeFallbackQueryPendingTransaction` for handling fallback query transactions.

9bfa71dd039ac1936670f19b5603934b78c44729: [synapse-interface preview link ](https://sanguine-synapse-interface-r6cwmsjlk-synapsecns.vercel.app)
1266f3de88901f551e32aa0cd7113195ed1f713f: [synapse-interface preview link ](https://sanguine-synapse-interface-69udxwe9o-synapsecns.vercel.app)
c88049e7514aab801312d9a37b120936c5f094db: [synapse-interface preview link ](https://sanguine-synapse-interface-i9axox2a2-synapsecns.vercel.app)
8d4cf763eec628b45b3b4858b15ed28e68255605: [synapse-interface preview link ](https://sanguine-synapse-interface-lvxzik0bo-synapsecns.vercel.app)
c5322e2165ef35e720ca52a76ead6c8d86e0cd5e: [synapse-interface preview link ](https://sanguine-synapse-interface-69l8hm5p1-synapsecns.vercel.app)
b034775e41ae4fb3ee1e720d27511299fcc30656: [synapse-interface preview link ](https://sanguine-synapse-interface-ila5d4jig-synapsecns.vercel.app)
2cea4140fb9f3ffa6e9c523dab0cd0edd7000a60: [synapse-interface preview link ](https://sanguine-synapse-interface-25f5i7lnf-synapsecns.vercel.app)
0b09ab4cf70f298a5a2186570667b0e06f2cfc2e: [synapse-interface preview link ](https://sanguine-synapse-interface-5dwobr5ee-synapsecns.vercel.app)
8658123a367fe3438548ee0aeaf1d949925eaf75: [synapse-interface preview link ](https://sanguine-synapse-interface-a6vpiyr2t-synapsecns.vercel.app)
3f4dc92804d8b60ec63e2890ccbb8baad40bbe99: [synapse-interface preview link ](https://sanguine-synapse-interface-1n66zc2a9-synapsecns.vercel.app)
59b4dff384b1d4b0b57535b5692d9a1126750cb8: [synapse-interface preview link ](https://sanguine-synapse-interface-g9ys8j6kj-synapsecns.vercel.app)
d1f968d3fca9746f791e7c7829de7be389e682c4: [synapse-interface preview link ](https://sanguine-synapse-interface-cqlrpvy9h-synapsecns.vercel.app)
6218f3bdea36d572c967cd0da0a665a9a6ba8c0c: [synapse-interface preview link ](https://sanguine-synapse-interface-bwwujmpj1-synapsecns.vercel.app)
7ce44a1abc2c0f92c583258899f4726d134c1fcb: [synapse-interface preview link ](https://sanguine-synapse-interface-1xwnms6wn-synapsecns.vercel.app)
0f085ee5160e1e7e98dbb9b4f6b3ec338667ea76: [synapse-interface preview link ](https://sanguine-synapse-interface-jgb71e1jk-synapsecns.vercel.app)
4ecff12d592f8c5a01e9c2d5d1784ee39147b4ea: [synapse-interface preview link ](https://sanguine-synapse-interface-kzwkf4v7n-synapsecns.vercel.app)
c6c25cdb42608d1e5fe89c4050a417e5d4c20902: [synapse-interface preview link ](https://sanguine-synapse-interface-bkgajicoc-synapsecns.vercel.app)
970880b81e8b7fc7216c6384b09c8d9b0b64b47a: [synapse-interface preview link ](https://sanguine-synapse-interface-m638omjwr-synapsecns.vercel.app)
ce298f531ad92f11d8bfb719dc7358ff46581866: [synapse-interface preview link ](https://sanguine-synapse-interface-ezkclgop8-synapsecns.vercel.app)
77945119ec3098c18305ce91ba28f0bfbb92a6b4: [synapse-interface preview link ](https://sanguine-synapse-interface-e7mb3eq2f-synapsecns.vercel.app)
cf13df00eaca4582fd0b1d2fa3cee3db9dc0b2f9: [synapse-interface preview link ](https://sanguine-synapse-interface-m68p3jgl3-synapsecns.vercel.app)
5921e33bf33ed7589b6e3c7a73f87d9ebdba7590: [synapse-interface preview link ](https://sanguine-synapse-interface-aow3e55l5-synapsecns.vercel.app)
0d16ad68652e54234cdbf929190a3d495bc11085: [synapse-interface preview link ](https://sanguine-synapse-interface-kr443ubzd-synapsecns.vercel.app)
22fff18efb3c8cdaa4a463b2d346663aca3857c5: [synapse-interface preview link ](https://sanguine-synapse-interface-ccu67sait-synapsecns.vercel.app)
9c8767eafe8a9229bc6770758e9af580b8661b4b: [synapse-interface preview link ](https://sanguine-synapse-interface-gk3qafvfz-synapsecns.vercel.app)
224c70d89f596e3081abf6b5d757a7c07b1f4839: [synapse-interface preview link ](https://sanguine-synapse-interface-7g4dsjbde-synapsecns.vercel.app)
b7ca042c3095719274bf428b91e28f5a44cb0a9d: [synapse-interface preview link ](https://sanguine-synapse-interface-6u32olq3s-synapsecns.vercel.app)
ebd8af2c26ce3ef0e0954304f90c68db8db17815: [synapse-interface preview link ](https://sanguine-synapse-interface-q8z1g1te7-synapsecns.vercel.app)
cfdc670268acad7728258819d7e5f52588619362: [synapse-interface preview link ](https://sanguine-synapse-interface-9zsd6snqp-synapsecns.vercel.app)
0898740e8bf16354a2992df46a648c37b945e214: [synapse-interface preview link ](https://sanguine-synapse-interface-e7fsgkrtl-synapsecns.vercel.app)
58da47faccef0751fb3bdb4404fcc3b6ba9b3000: [synapse-interface preview link ](https://sanguine-synapse-interface-ejzmw5qa3-synapsecns.vercel.app)
0481b7d8d3a9eccffcc7e9ffb48bc72a6413336f: [synapse-interface preview link ](https://sanguine-synapse-interface-pvr3r2eob-synapsecns.vercel.app)
a7842fc0772643cbf760e1b249ad384ba2864dd1: [synapse-interface preview link ](https://sanguine-synapse-interface-h8m3cuccf-synapsecns.vercel.app)
4f07acf1574c42b9cf18e98b68daeb5835c4285a: [synapse-interface preview link ](https://sanguine-synapse-interface-hw3lf1ns8-synapsecns.vercel.app)
9542373d234487923b5d2d49a06be1d1a282af04: [synapse-interface preview link ](https://sanguine-synapse-interface-h93b73jv7-synapsecns.vercel.app)
8ffc02f8b3e905d673530a93881423bd81262f39: [synapse-interface preview link ](https://sanguine-synapse-interface-68udh4c1y-synapsecns.vercel.app)
5dfbfe27bddb7e7d21a7263f70aa78415a0212fb: [synapse-interface preview link ](https://sanguine-synapse-interface-2j8y6xi3k-synapsecns.vercel.app)
5c77cddba819a732fe561f03b5abc499bd8480f4: [synapse-interface preview link ](https://sanguine-synapse-interface-j2itouk1y-synapsecns.vercel.app)

### CODE REVIEW

File: packages/synapse-interface/components/StateManagedBridge/BridgeExchangeRateInfo.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ const BridgeExchangeRateInfo = ({ showGasDrop }: { showGasDrop: boolean }) => {
   const toChainId = useSelector((state: RootState) => state.bridge.toChainId)
   const { gasDrop: gasDropAmount, loading } = useGasDropAmount(toChainId)
 
-  const safeExchangeRate = exchangeRate ?? 0n
+  const safeExchangeRate = BigInt(exchangeRate) ?? 0n
```
[the remark this answers is not in the document]
Comment (bigboydiamonds) [reply packages/synapse-interface/components/StateManagedBridge/BridgeExchangeRateInfo.tsx]: Updated

File: packages/synapse-interface/slices/transactions/actions.ts [hunk 2]
```diff
@@ -24,6 +24,25 @@ export const addPendingAwaitingCompletionTransaction =
...
+  'transactions/removeFallbackQueryPendingTransaction'
+)
+export const addFallbackQueryHistoricalTransaction =
+  createAction<BridgeTransaction>(
+    'transaction/addFallbackQueryHistoricalTransaction'
+  )
+export const removeFallbackQueryHistoricalTransaction =
+  createAction<BridgeTransaction>(
+    'transaction/removeFallbackQueryHistoricalTransaction'
+  )
```
[the remark this answers is not in the document]
Comment (bigboydiamonds) [reply packages/synapse-interface/slices/transactions/actions.ts:45]: Updated

File: packages/synapse-interface/slices/transactions/updater.tsx [hunk 3]
```diff
@@ -40,6 +43,7 @@ import {
   removePendingAwaitingCompletionTransaction,
 } from './actions'
 import { getValidAddress } from '@/utils/isValidAddress'
+import { checkTransactionsExist } from '@/components/Portfolio/Activity'
```
[the remark this answers is not in the document]
Comment (bigboydiamonds) [reply packages/synapse-interface/slices/transactions/updater.tsx]: Updated

**Review by aureliusbtc — APPROVED** [review]

**Comment by bigboydiamonds** [comment]
Upgrade `viem` `wagmi` `@wagmi/core` to latest versions as of 10.30.2023

**Comment by bigboydiamonds** [comment]
@trajan0x added in ability to reset `redux-persist` cache based on valid version denoted in `persistConfig`: 

https://github.com/synapsecns/sanguine/pull/1474/commits/92664fd87e28fca28daf00af5ee3c9c9ca89e6b5

**Comment by bigboydiamonds** [comment]
@coderabbitai pause

**Pull request merged**